File: example2/main.py
from wasmtime import Store, Module, Instance # type: ignore

store = Store()
module = Module.from_file(store.engine, 'sum.wasm')
imports = []
instance = Instance(store, module, imports)
exports = instance.exports(store)
run = exports["sum"]
result = run(store, 20, 22)
print(result)

File: example2/main.rs
use wasmtime::*;

fn main() -> Result<()> {
    let engine = Engine::default();
    let module = Module::from_file(&engine, "sum.wasm")?;
    let mut store = Store::new(&engine, ());

    let imports = [];
    let instance = Instance::new(&mut store, &module, &imports)?;

    let sum = instance.get_typed_func::<(i32, i32), i32>(&mut store, "sum")?;
    let result= sum.call(&mut store, (20, 22))?;
    println!("{result}");

    Ok(())
}
